apis/v1/gateway_types.go

```diff
@@ -188,6 +188,7 @@ type GatewaySpec struct {
 	// +kubebuilder:validation:MaxItems=64
 	// +kubebuilder:validation:XValidation:message="tls must be specified for protocols ['HTTPS', 'TLS']",rule="self.all(l, l.protocol in ['HTTPS', 'TLS'] ? has(l.tls) : true)"
 	// +kubebuilder:validation:XValidation:message="tls must not be specified for protocols ['HTTP', 'TCP', 'UDP']",rule="self.all(l, l.protocol in ['HTTP', 'TCP', 'UDP'] ? !has(l.tls) : true)"
+	// +kubebuilder:validation:XValidation:message="tls mode must be Terminate for protocol HTTPS",rule="self.all(l, l.protocol == 'HTTPS' ? (l.tls.mode == '' || l.tls.mode == 'Terminate') : true)"
```

Comment:
Maybe we can also verify if `l.tls` actually exists to be more precise on when the error is returned.

```
l.protocol == 'HTTPS' && has(l.tls)
```

Reply:
@zhaohuabing -- Thanks for the update! The behaviour that I had in mind was to show "one exact error":
```
(l.protocol == 'HTTPS' && has(l.tls)) ? (l.tls.mode == '' || l.tls.mode == 'Terminate') : true 
```

As opposed to the following which will also display `tls mode must be Terminate for protocol HTTPS`, when the `tls` field is missing:
```
l.protocol == 'HTTPS' ? (has(l.tls) && (l.tls.mode == '' || l.tls.mode == 'Terminate')) : true
```